File: csharp/6-kyu/number-zoo-patrol/solution.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Kata
{
  // Return the missing number!
  public static int FindNumber(int[] array)
  {
    if (array.Length < 1)
    {
      return 1;
    }
    else
    {
      var missing = Enumerable.Range(1, array.Max()).Except(array);
      return missing.Any() ? missing.First() : array.Max() + 1;
    }
  }
}

File: javascript/6-kyu/number-zoo-patrol/solution.js
// return the missing number!
function findNumber(array) {
  let sum = array.length + 1;
  let expectedCount = (sum * (sum + 1)) / 2;
  let actualCount = array.reduce((a, b) => a + b, 0);
  return expectedCount - actualCount;
}
